pytorch_blade/tests/mhlo/matmul.mlir

```diff
@@ -162,3 +162,44 @@ func.func @torch.aten.mm.proj(%arg0: !torch.vtensor<[?,256],f32>) -> !torch.vten
   return %1 : !torch.vtensor<[?,256],f32>
 }
 
+// -----
+
+// CHECK-LABEL:  func.func @torch.aten.matmul.dynamic_shape_cast(
+// CHECK-SAME:         %[[ARG0:.*]]: tensor<2x?x?xf32>) -> tensor<2x256x256xf32> {
+// CHECK:         %[[T0:.*]] = mhlo.constant dense<1.000000e+00> : tensor<2x256x256xf32>
+// CHECK:         %[[T1:.*]] = tensor.cast %[[ARG0]] : tensor<2x?x?xf32> to tensor<2x?x256xf32>
+// CHECK:         %[[T2:.*]] = "mhlo.dot_general"(%[[T1]], %[[T0]]) {dot_dimension_numbers = #mhlo.dot<lhs_batching_dimensions = [0], rhs_batching_dimensions = [0], lhs_contracting_dimensions = [2], rhs_contracting_dimensions = [1]>} : (tensor<2x?x256xf32>, tensor<2x256x256xf32>) -> tensor<2x?x256xf32>
+// CHECK:         %[[T3:.*]] = mhlo.convert(%[[T2]]) : (tensor<2x?x256xf32>) -> tensor<2x256x256xf32>
```

Comment:
using `tensor.cast` here?

Reply:
The convert is derived from `llvm/torch-mlir`, I would not fix it in our repo.